services/src/main/java/org/keycloak/services/clienttype/impl/DefaultClientType.java

```diff
@@ -73,4 +75,9 @@ public Map<String, ClientTypeRepresentation.PropertyConfig> getConfiguration() {
     public ClientModel augment(ClientModel client) {
         return new TypeAwareClientModelDelegate(this, () -> client);
     }
+
+    @Override
+    public ClientRepresentation augment(ClientRepresentation representation) {
+        return new TypeDefaultedClientRepresentation(this, representation);
```

Comment:
Yeah that would be ideal.

What I am struggling to solve is that some of the getters on ClientModel are returning primitives, for instance, `isPublicClient` which returns boolean. We cannot know whether the user had specified some value or not within `TypeAwareClientModelDelegate`. That means we cannot determine whether to return the client type default value (specifically for non-read only fields) or what is currently set on the delegate.
* https://github.com/keycloak/keycloak/blob/main/services/src/main/java/org/keycloak/services/clienttype/client/TypedClientSimpleAttribute.java#L112

Since the `TypeAwareClientModelDelegate` class is the only thing that is currently client type aware, it seemed like we would have to make something else type aware to satisfy the ability to support client type fields with override-able (non-read only) default values. That was the purpose of the `TypeDefaultedClientRepresentation` which is currently only applied during client creation. It's purpose is to return the field supplied by the user or if not supplied, return the default value on the client type.

There was some other things we could consider:
* Make these methods return non-primitive alternatives.
  * Con: These are public interfaces, would be a large breaking change.
  * These fields are non-null in the db so they would have to be set before the entity is persisted. So this solution might not be possible at all.
* Make client type field values read-only.
 * Con: Cannot override values. Admin has to create a new client type definition.

Open to suggestions on what you think.
cc @vickeybrown @jsorah